README.markdown

```diff
@@ -6884,6 +6888,26 @@ See also [ngx.shared.DICT](#ngxshareddict).
 
 [Back to TOC](#nginx-api-for-lua)
 
+ngx.shared.DICT.cog
+-------------------
+**syntax:** *value, flags = ngx.shared.DICT:cog(key, old_value?, old_flags?)*
+
+**context:** *set_by_lua&#42;, rewrite_by_lua&#42;, access_by_lua&#42;, content_by_lua&#42;, header_filter_by_lua&#42;, body_filter_by_lua&#42;, log_by_lua&#42;, ngx.timer.&#42;, balancer_by_lua&#42;, ssl_certificate_by_lua&#42;, ssl_session_fetch_by_lua&#42;, ssl_session_store_by_lua&#42;, ssl_client_hello_by_lua&#42;*
+
+Similar to the [get](#ngxshareddictget) method, but only returns if
```

Comment:
What does the "only returns if" part mean? If the if condition is not met, this method will never return? It cannot be. Maybe you mean "will only return the value and the flags" here? And you should also be explicit about what values it returns otherwise (like two nil values?)

The `cog` name is a bit confusing. How about `get_if_not_eq`?

Reply:
Yeah the "only returns if" part is not true. 
Maybe something like:
```
Similar to the [get](#ngxshareddictget) method, but if ``old_value`` and
``old_flags`` match value and flag in dictionary ngx.shared.DICT, 
returns ``nil, true``.
```
I'm fine with ``get_if_not_eq``